hotels/views.py

```diff
@@ -1,3 +1,114 @@
+from django.db.models       import Q, Avg, Min
+
+from hotels.models import Category, Hotel, Room
+
+class CategoryLocationView(View):
+    def get(self, request):
+        try:
+            results = [
+                    {
+                        'location'  : [location.name for location in category.location.all()],
+                        'category'  : category.name,
+                        'image_url' : category.thumbnail_image
+                        }for category in Category.objects.all().prefetch_related('location')]
+
+            return JsonResponse({'MESSAGE':'SUCCESS', 'results':results}, status=200)
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
+
+class HotelView(View):
+    def get(self, request):
+        try:
+            category_name = request.GET.get('category_name')
+            location_name = request.GET.get('location_name')
+            occupancy     = request.GET.get('occupancy')
+            check_in      = request.GET.get('check_in')
+            check_out     = request.GET.get('check_out')
+            sort_type     = request.GET.get('sort_type')
+            search        = request.GET.get('search')
+
+            hotel_lists = Hotel.objects.filter(
+                    Q(category__name       = category_name) &
+                    Q(location__name       = location_name) &
+                    Q(room__occupancy__gte = occupancy) &
+                    (
+                        Q(room__reservationcheck__date__gte = check_in) & 
+                        Q(room__reservationcheck__date__lt  = check_out)
+                        )
+                    ).distinct()
+            
+            if sort_type == '-rate':
+                    hotel_lists = hotel_lists.annotate(rate = Avg('review__rate')).order_by('-rate')
+            
+            hotel_lists = hotel_lists.annotate(price = Min('room__original_price')).order_by(sort_type)
+
+            if search:
+                hotel_lists = Hotel.objects.filter(name__icontains = search)
+
+            if check_in >= check_out:
+                return JsonResponse({'MESSAGE':'INVALID_DATE'}, status=400)
+
+            if int(occupancy) >= 5:
+                return JsonResponse({'MESSAGE':'INVALIED_OCCUPANCY'}, status=400)
+
+            
+            results = [
+                    {
+                        'id'                : hotel.id,
+                        'name'              : hotel.name,
+                        'address'           : hotel.address,
+                        'thumbnail_image'   : hotel.thumbnail_image,
+                        'star'              : hotel.star,
+                        'original_price'    : int(hotel.room_set.all().order_by('original_price').first().original_price),
+                        'discount_price'    : int(hotel.room_set.all().order_by('original_price').first().original_price * hotel.room_set.all().order_by('original_price').first().discount_rate),
```

Comment:
여긴 room과 관련된 부분 이죠?
구분해줘야 할 것 같습니다!
지난 리뷰에서 room 과 hotel 을 구분해서 return 할 필요가 없다고 한 피드백은
{'MESSAGE' : 'SUCCESS', 'hotels' : hotel, 'rooms' : room } 과 같이 구분하는 건 불필요하다고 한 것이고,
지금과 같은 경우는 아래와 같이 구분은 해줘야 합니다!
```suggestion
        'room' : room_list 
        hotel.room_set.all().order_by('original_price').first().discount_rate),
```

Reply:
네이밍 변경하겠습니다~!